.github/workflows/test.yml

```diff
@@ -1,52 +1,38 @@
-## test.yml
+name: CI
 
-name: Main workflow
-on: [push, pull_request]
+on:
+  pull_request:
+    paths-ignore:
+    - '**.md'
+    - '**.rst'
+  push:
+    paths-ignore:
+    - '**.md'
+    - '**.rst'
+    branches-ignore:
+    - 'master'
+    - 'main'
 
 jobs:
   build:
     runs-on: ubuntu-latest
     strategy:
-      fail-fast: false
       matrix:
-        emacs_version:
-          - '24.1'
-          - '24.2'
-          - '24.3'
-          - '24.4'
-          - '24.5'
-          - '25.1'
-          - '25.2'
-          - '25.3'
-          - '26.1'
-          - '26.2'
-          - '26.3'
-          - '27.1'
-          - 'snapshot'
-        include:
-          - emacs_version: 'snapshot'
-            allow_failure: true
+        os: [ubuntu-latest]
+        python_version: [3.6]
+        emacs_version: [24.4, 24.5, 25.3, 26.3, 27.1]
```

Comment:
Please check every Emacs-24.1~Emacs27.1 and snapshot.
If you want to remove this (I think so too), we should finish the #523 discussion.

Reply:
Addressed 37052df